src/Analyzers/MSTest.Analyzers/DuplicateDataRowAnalyzer.cs

```diff
@@ -0,0 +1,139 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Immutable;
+
+using Analyzer.Utilities.Extensions;
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.Testing.Platform;
+
+using MSTest.Analyzers.Helpers;
+
+namespace MSTest.Analyzers;
+
+/// <summary>
+/// MSTEST0042: <inheritdoc cref="Resources.DuplicateDataRowTitle"/>.
+/// </summary>
+[DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+public sealed class DuplicateDataRowAnalyzer : DiagnosticAnalyzer
+{
+    internal static readonly DiagnosticDescriptor Rule = DiagnosticDescriptorHelper.Create(
+        DiagnosticIds.DuplicateDataRowRuleId,
+        new LocalizableResourceString(nameof(Resources.DuplicateDataRowTitle), Resources.ResourceManager, typeof(Resources)),
+        new LocalizableResourceString(nameof(Resources.DuplicateDataRowMessageFormat), Resources.ResourceManager, typeof(Resources)),
+        null,
+        Category.Usage,
+        DiagnosticSeverity.Warning,
+        isEnabledByDefault: true);
+
+    /// <inheritdoc />
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
+
+    /// <inheritdoc />
+    public override void Initialize(AnalysisContext context)
+    {
+        context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+        context.EnableConcurrentExecution();
+
+        context.RegisterCompilationStartAction(context =>
+        {
+            if (context.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.MicrosoftVisualStudioTestToolsUnitTestingDataRowAttribute, out INamedTypeSymbol? dataRowAttribute))
+            {
+                context.RegisterSymbolAction(
+                    context => AnalyzeSymbol(context, dataRowAttribute),
+                    SymbolKind.Method);
+            }
+        });
+    }
+
+    private static void AnalyzeSymbol(SymbolAnalysisContext context, INamedTypeSymbol dataRowAttribute)
+    {
+        var methodSymbol = (IMethodSymbol)context.Symbol;
+        var dataRowArguments = new HashSet<ImmutableArray<TypedConstant>>(TypedConstantArrayComparer.Instance);
+
+        foreach (AttributeData attribute in methodSymbol.GetAttributes())
+        {
+            if (!dataRowAttribute.Equals(attribute.AttributeClass, SymbolEqualityComparer.Default))
+            {
+                continue;
+            }
+
+            if (!dataRowArguments.Add(attribute.ConstructorArguments))
+            {
+                context.ReportDiagnostic((attribute.ApplicationSyntaxReference?.GetSyntax(context.CancellationToken).GetLocation() ?? Location.None).CreateDiagnostic(Rule));
```

Comment:
Is there no helper for that on roslyn-analyzers? Or maybe it's already available in the code I copied.